test/stream.test.ts

```diff
@@ -72,6 +72,30 @@ describe('DataAndMoneyStream', function () {
       const clientStream = this.clientConn.createStream()
       assert.throws(() => clientStream.setSendMax(Infinity), 'sendMax must be finite')
     })
+
+    it('should throw if the stream is in receiveOnly mode', async function () {
```

Comment:
Its probably good to duplicate this test in the opposite direction, client with receiveOnly=true.